diffrax/_solver/srk.py

```diff
@@ -0,0 +1,624 @@
+            assert len(self.ignore_stage_f) == len(self.b_sol)
+        if self.ignore_stage_g is not None:
+            assert len(self.ignore_stage_g) == len(self.b_sol)
+
+
+StochasticButcherTableau.__init__.__doc__ = """**Arguments:**
+
+Let `k` denote the number of stages of the solver.
+
+- `a`: the lower triangle (without the diagonal) of the Butcher tableau. Should
+    be a tuple of NumPy arrays, corresponding to the rows of this lower triangle. The
+    first array represents the should be of shape `(1,)`. Each subsequent array should
+    be of shape `(2,)`, `(3,)` etc. The final array should have shape `(k - 1,)`.
+- `b_sol`: the linear combination of stages to take to produce the output at each step.
+    Should be a NumPy array of shape `(k,)`.
+- `b_error`: the linear combination of stages to take to produce the error estimate at
+    each step. Should be a NumPy array of shape `(k,)`. Note that this is *not*
+    differenced against `b_sol` prior to evaluation. (i.e. `b_error` gives the linear
+    combination for producing the error estimate directly, not for producing some
+    alternate solution that is compared against the main solution).
+- `c`: the time increments used in the Butcher tableau.
+    Should be a NumPy array of shape `(k-1,)`, as the first stage has time increment 0
+- `cW`: The coefficients in front of the Brownian increment at each stage.
+    Should be a NumPy array of shape `(k,)`. Only used when `additive_noise=True`.
+- `cH`: The coefficients in front of the space-time Lévy area at each stage.
+    Should be a NumPy array of shape `(k,)`. Only used when `additive_noise=True`.
+- `bW`: Coefficient for the Brownian increment when computing the
+    output $y_{n+1}$. Should be a `Scalar`.
+- `bH`: Coefficient for the space-time Lévy area when computing the
+    output $y_{n+1}$. Should be a `Scalar`.
+- `aW`: The coefficients in front of the Brownian increment at each stage.
+    Should be of the same shape as `a`. Only used when `additive_noise=False`.
+- `aH`: The coefficients in front of the space-time Lévy area at each stage.
+    Should be of the same shape as `a`. Only used when `additive_noise=False`.
+- `additive_noise`: Whether the SDE has additive noise. If `True`, then `bW` and `bH`
+    should be scalars, and `cW` and `cH` should be arrays of shape `(k,)`. If `False`,
+    then `bW` and `bH` should be arrays of shape `(k,)`, `cW` and `cH` should be `None`,
+    and `aW` and `aH` should be lists of arrays of the same shape as `a`.
+"""
+
+
+class AbstractSRK(AbstractSolver[_SolverState]):
+    r"""A general Stochastic Runge-Kutta method.
+
+    The second term in the MultiTerm must be a `ControlTerm` with
+    `control=VirtualBrownianTree`. Depending on the Butcher tableau, the
+    `VirtualBrownianTree` may need to be initialised with 'levy_area="space-time"'
+    or 'levy_area="space-time-time"'.
+
+    Given the Stratonovich SDE
+    $dX_t = f(t, X_t) dt + g(t, X_t) \circ dW_t$
```

Comment:
Nit: the convention I use throughout Diffrax is for the evolving solution of any SDE to be represented as (lower case) `y`, and for Brownian motions to be (lower case) `w`. Likewise, evaluation of a path at a particular time is done using bracket notation, e.g. `dw(t)`, not subscript notation.

If you're curious, I standardised on this whilst writing my thesis back in the day -- I found that these choices are part of being able to write all of ODEs/SDEs/CDEs using the exact same notation in all circumstances.

Reply:
Sounds good. I'll try to find all the docstrings where this appears.